library(tidyverse)
#Read ExAC variant filtered by Thomas (AF > 1e-5), lacking chr5 and chr15
Ex <- data.table::fread("ExAC_all_snp.csv", header = FALSE)
Ex <- Ex[-1,]
Ex <- separate(data = Ex, col = V2, into = letters[1:14], sep = ",")
head(Ex)
Ex <- Ex %>% select(c(b,c,e,f)) %>% 
  unite("Genomic_Position", c("b","c"), sep = ":", remove = TRUE, na.rm = FALSE) %>%
  unite("SNV", c("e", "f"), sep = ">", remove = TRUE, na.rm = FALSE) %>%
  unite("Genomic_Change", c("Genomic_Position", "SNV"), sep = "", remove = FALSE, na.rm = FALSE) %>%
  select(c(Genomic_Change, Genomic_Position))

#Read chr5 and chr15 file generated by Thomas and filter out AF >= 1e-5
chr15 <- read.table(text = gsub("=", "\t", readLines("chr15_SNPs_AFtrimmed.txt")))
chr15 <- chr15 %>% select(c(V1,V2,V4,V5,V13)) %>% 
  filter(V13 > 1e-5 | V13 == 1e-5) %>%
  unite("Genomic_Position", c("V1","V2"), sep = ":", remove = TRUE, na.rm = FALSE) %>%
  unite("SNV", c("V4", "V5"), sep = ">", remove = TRUE, na.rm = FALSE) %>%
  unite("Genomic_Change", c("Genomic_Position", "SNV"), sep = "", remove = FALSE, na.rm = FALSE) %>%
  select(c(Genomic_Change, Genomic_Position))

chr5 <- read.table(text = gsub("=", "\t", readLines("chr5_SNPs_AFtrimmed.txt")))
chr5 <- chr5 %>% select(c(V1,V2,V4,V5,V13)) %>%
  filter(V13 > 1e-5 | V13 == 1e-5) %>%
  unite("Genomic_Position", c("V1","V2"), sep = ":", remove = TRUE, na.rm = FALSE) %>%
  unite("SNV", c("V4", "V5"), sep = ">", remove = TRUE, na.rm = FALSE) %>%
  unite("Genomic_Change", c("Genomic_Position", "SNV"), sep = "", remove = FALSE, na.rm = FALSE) %>%
  select(c(Genomic_Change, Genomic_Position))

#Combine Ex and chr5 and chr15 to generate a new Ex(ExAC variant) with all chromosomes
Ex <- rbind(Ex, chr5, chr15)


